pkg/operator/hive/hive_controller.go

```diff
@@ -337,6 +353,54 @@ func (r *ReconcileHiveConfig) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileHiveConfig) establishSecretWatch(hLog *log.Entry, hiveNSName string) error {
+	// We need to establish a watch on Secret in the Hive namespace, one time only. We do not know this namespace until
+	// we have a HiveConfig.
+	if !r.servingCertSecretWatchEstablished {
+		hLog.WithField("namespace", hiveNSName).Info("establishing watch on secrets in hive namespace")
+
+		// Create an informer that only listens to events in the OpenShift managed namespace
+		kubeInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(r.kubeClient, 10*time.Minute,
+			kubeinformers.WithNamespace(hiveNSName))
+		secretsInformer := kubeInformerFactory.Core().V1().Secrets().Informer()
+		err := r.mgr.Add(&informerRunnable{informer: secretsInformer})
+		if err != nil {
+			hLog.WithError(err).Error("error adding secret informer to manager")
+			return err
+		}
+
+		// Watch Secrets in hive namespace, so we can detect changes to the hiveadmission serving cert secret and
+		// force a deployment rollout.
+		err = r.ctrlr.Watch(&source.Informer{Informer: secretsInformer}, handler.Funcs{
+			CreateFunc: func(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+				log.Debug("eventHandler CreateFunc")
```

Comment:
Should this use `hLog` instead of `log`? It would also be nice if the message were a bit more descriptive--at least mentioning that it is the event handler for the secret watcher.

Reply:
Yup another remnant from when it was outside the reconciler. Updated.